crates/core/src/client/config.rs

```diff
@@ -61,19 +67,35 @@ pub struct StarknetConfig {
     pub kakarot_address: FieldElement,
     /// Proxy account class hash.
     pub proxy_account_class_hash: FieldElement,
+    /// EOA class hash.
+    pub externally_owned_account_class_hash: FieldElement,
+    /// Contract Account class hash.
+    pub contract_account_class_hash: FieldElement,
 }
 
 impl StarknetConfig {
-    pub fn new(network: Network, kakarot_address: FieldElement, proxy_account_class_hash: FieldElement) -> Self {
-        StarknetConfig { network, kakarot_address, proxy_account_class_hash }
+    pub fn new(
+        network: Network,
+        kakarot_address: FieldElement,
+        proxy_account_class_hash: FieldElement,
+        externally_owned_account_class_hash: FieldElement,
+        contract_account_class_hash: FieldElement,
+    ) -> Self {
+        StarknetConfig {
+            network,
+            kakarot_address,
+            proxy_account_class_hash,
+            externally_owned_account_class_hash,
+            contract_account_class_hash,
+        }
```

Comment:
though I do see in some rust example this `new` function that does nothing more than creating the struct with all the param, I still don't understand why we do so. any hint?

Reply:
iirc it is if struct fields are private then you need to use new or default

Reply:
adding to @Eikix, it can also be for encapsulation, from my understanding. Even with public fields, if you have a `new` function to create instances, you can abstract away the defails of instantiation i.e. can change the internal details without changing the API